src/Signer.php

```diff
@@ -39,6 +42,39 @@ public function sign($message)
         return $message;
     }
 
+    /**
+     * @param RequestInterface $message
+     * @return RequestInterface
+     */
+    public function signWithDigest($message)
+    {
+        $message = $this->addDigest($message);
+        return $this->sign($message);
+    }
+
+    /**
+     * @param RequestInterface $message
+     * @return RequestInterface
+     */
+    private function addDigest($message)
+    {
+        if (!array_search('digest', $this->headerList->names)) {
+            $this->headerList->names[] = 'digest';
+        };
+        while ($message->getHeader('Digest')) {
+            $message = $message->withoutHeader('Digest');
+        };
+        $message = $message->withHeader(
+        'Digest',
+        self::defaultHttpDigestPrefix . '=' . base64_encode(
+          hash(
+            self::defaultHttpDigest, $message->getBody(), true
+          )
+        )
+      );
```

Comment:
I would suggest inlining the consts here to simplify - they're only referenced here, so keeping it together in this case would be simpler

Reply:
Resolved in 25727bb93fd77099e3cc561cb6ee384cde314ea2